### PR TITLE
[Flow] Avoid collapsing reduction dimensions on pre-formed dispatches

### PR DESCRIPTION
**Pull request by Max191**
Rebased on top of #14433

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/test/collapse_reduction.mlir [hunk 1]
```diff
@@ -37,3 +37,28 @@ func.func @input_broadcast(%arg0: tensor<4x8xf32>, %arg1: tensor<4xf32>) -> tens
...
+    } -> tensor<2x32xf32>
+    flow.return %6 : tensor<2x32xf32>
+  }
+  %4 = tensor.expand_shape %3 [[0], [1, 2, 3]] : tensor<2x32xf32> into tensor<2x32x1x1xf32>
+  %5 = hal.tensor.export %4 : tensor<2x32x1x1xf32> -> !hal.buffer_view
+  return %5 : !hal.buffer_view
+}
+
+// CHECK: @multi_reduce_dim_dispatch
+// CHECK: linalg.generic {{.*}} iterator_types = ["parallel", "parallel", "reduction", "reduction"]
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/test/collapse_reduction.mlir]: Nit add new line at end of file.

**Review by MaheshRavishankar — APPROVED** [review]
Looks good. Will stamp, but please wait for the base PR to land.

**Comment by qedawkins** [comment]
#14433 has landed so you can rebase on top of main.

**Comment by Max191** [comment]
Rebased on top of main

**Comment by MaheshRavishankar** [comment]
Something is off with the rebase. You have commits that are not part of the PR

**Comment by Max191** [comment]
Rebase seems right now

**Pull request merged**